homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,428 @@
+                    colortemp,
+                    sceneid,
+                )
+
+            sceneid = None
+            if ATTR_EFFECT in kwargs:
+                sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+            if sceneid == 1000:  # rhythm
+                pilot = PilotBuilder()
+            else:
+                pilot = PilotBuilder(
+                    brightness=brightness, colortemp=colortemp, scene=sceneid
+                )
+
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.max
+            )
+
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 6500")
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            return self.featuremap()
+        # fallback
+        return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+            await self.update_scene_list()
+
+    @property
+    def device_info(self):
```

Comment:
This could be set once as `self._attr_device_info` in `__init__` to avoid it happening each time the state is updated